templates/general/ssl.j2

```diff
@@ -8,4 +8,5 @@
   ssl_protocols {{ssl_protocols}};
   ssl_dhparam '{{ssl_dh_file}}';
   ssl_prefer_server_ciphers on;
-  ssl_session_cache shared:SSL:10m;
\ No newline at end of file
+  ssl_session_cache shared:SSL:10m;
+  add_header Strict-Transport-Security max-age=15768000;
```

Comment:
👍 for HSTS support!

The header should only be sent when encryption mode is `force` or `redirect`, but not `optional`. I wonder what the best role interface is to enable HSTS and also tweak the max-age value. For testing/development purposes a week might be a suitable timeout, while in production environments it might years.

A new role variable `ssl_hsts_max_age`, defaulting to `604800` seconds (1 week), could suppress the header when being set to `0`, `false` or `no` to disable HSTS entirely. Probably I would not dare a higher default timeout, because an accident in the configuration could have consequences for a very long time.

What's your opinion on the default max age value and the single integer variable to disable HSTS when set to `0`. Is that intuitive to use or is it better to have a boolean toggle like `ssl_hsts_enabled`?